prometheus/promhttp/instrument_server.go

```diff
@@ -348,15 +394,27 @@ func sanitizeMethod(m string) string {
 		return "options"
 	case "NOTIFY", "notify":
 		return "notify"
+	case "TRACE", "trace":
+		return "trace"
+	case "PATCH", "patch":
+		return "patch"
 	default:
-		return strings.ToLower(m)
+		if len(additionalMethods) > 0 {
```

Comment:
Does this `if` really improve the performance? I would assume a `range` over an empty slice will exit very fast. But perhaps you know something I don't…

Reply:
Noop, you're it doesn't have any functional or performance effect. Just it was easier to reason about the implicit else clause `return "unknown"`, happy to remove and simplify.